packages/host/app/components/operator-mode/card-inheritance-panel.gts

```diff
@@ -0,0 +1,135 @@
+import Component from '@glimmer/component';
+import { service } from '@ember/service';
+import CardService from '@cardstack/host/services/card-service';
+import {
+  type CardDef,
+  type BaseDef,
+} from 'https://cardstack.com/base/card-api';
+import {
+  type RealmInfo,
+  cardTypeDisplayName,
+  identifyCard,
+  moduleFrom,
+} from '@cardstack/runtime-common';
+import DefinitionContainer, { DefinitionVariant } from './definition-container';
+import { isReady, FileResource } from '@cardstack/host/resources/file';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { ModuleSyntax } from '@cardstack/runtime-common/module-syntax';
+import LoaderService from '@cardstack/host/services/loader-service';
+import moment from 'moment';
+import { type ImportResource } from '@cardstack/host/resources/import';
+
+interface Args {
+  Element: HTMLElement;
+  Args: {
+    realmInfo: RealmInfo | null;
+    realmIconURL: string | null | undefined;
+    openFile: { current: FileResource | undefined };
+    cardInstance: CardDef | null;
+    importedModule?: ImportResource;
+    onCreate?: () => void;
+    onInherit?: () => void;
+    onDuplicate?: () => void;
+  };
+}
+
+export default class CardInheritancePanel extends Component<Args> {
+  @service declare cardService: CardService;
+  @service declare loaderService: LoaderService;
+  @tracked cardInstance: CardDef | undefined;
+  @tracked module: ModuleSyntax | undefined;
+
+  @action
+  duplicateAction() {
+    console.log('running duplicate');
+  }
+
+  @action
+  createAction() {
+    console.log('running create');
+  }
+
+  @action
+  inheritAction() {
+    console.log('running inherit');
+  }
+
+  <template>
+    <div class='container' ...attributes>
+      {{#if @importedModule.module}}
+        {{#each (cardsFromModule @importedModule.module) as |card|}}
+          <DefinitionContainer
+            @name={{this.getCardTypeDisplayName card}}
+            @fileExtension='.GTS'
```

Comment:
shouldn't this be parsed from the file name? we use .gts for cards right now, but we could also write cards as .ts or .gjs files instead.

Reply:
fair point! I think the machinery I have now to deduce a url from a module is insufficent. Since this case is for json instance

- I use the card resource in code mode to I have a json instance 
- I then identify the code-ref via `identifyCard`
- this yields `ref={module: 'http://localhost:4201/drafts/author', name: 'Author'}` 

wud like to handle this in another pr. Do you have a suggestion how to get this connection? I dont think we know the extension without literally knowing about the file.

https://linear.app/cardstack/issue/CS-5957/card-definition-needs-to-derive-the-file-type